### PR TITLE
Add temporary docs homepage

### PR DESCRIPTION
**Pull request by alexandratran**
Add temporary docs homepage as starting point for a more fleshed out developer hub homepage. This may help visualize what content and sections are possible and needed for the homepage.

Also fix issue with some links not consistently working (add trailingSlash rule and update some links' syntax).

Preview: https://docs.metamask.io/temp-homepage/

### CODE REVIEW

File: src/components/SnapsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    title: "⚙️⛽ Snaps tutorials",
+    link: "/snaps/category/tutorials",
+    description: (<>
+      Follow the step-by-step tutorials to create snaps that estimate gas fees and provide
+      transaction insights with custom UI.
+    </>),
+  },
+  {
+    title: "🌐 Snaps JSON-RPC API",
+    link: "/snaps/reference/cli",
```
Comment (bgravenorst) [inline src/components/SnapsSection.tsx:29]: This appears to link to the CLI, not APIs.

**Review by ziad-saab — CHANGES_REQUESTED** [review]
the temp homepage looks good! i have a few code wishes:

1. can the code in `custom.css` be modularized as `custom.module.css` as well?
2. can the `CardItem` type and the `Card` component, which are currently duplicated, be made into a custom component and re-used in both site sections

lmk what you think

**Review by ziad-saab — APPROVED** [review]

**Review by bgravenorst — APPROVED** [review]
LGTM. See minor suggestion.

**Review by vandan — APPROVED** [review]
Looks good to me from the perspective of having MVP content and a unified entry point for developers. Someone more technical should review the react code though.

**Review by shanejonas — APPROVED** [review]

**Pull request merged**